### ISSUE TITLE
current master version (from 2017-02-10) is not working

### ISSUE THREAD

**Issue by gerhard-tinned** [mozilla/ssh_scan#330]
Current master version (last commit e561c143f00fcaf936c8384c598742ff4b772b1f) does not work.

I cloned the repository, run the `bundle install` and tried to start the scanner which imidiatly showed the following error message.

<pre>
/usr/lib/ruby/2.3.0/rubygems/core_ext/kernel_require.rb:55:in `require': cannot load such file -- mongo (LoadError)
	from /usr/lib/ruby/2.3.0/rubygems/core_ext/kernel_require.rb:55:in `require'
	from /var/lib/gems/2.3.0/gems/backports-3.6.8/lib/backports/std_lib.rb:9:in `require_with_backports'
	from /home/gsteinbeis/security_monitor/3rd_party/ssh_scan/lib/ssh_scan/database/mongo.rb:1:in `<top (required)>'
	from /usr/lib/ruby/2.3.0/rubygems/core_ext/kernel_require.rb:55:in `require'
	from /usr/lib/ruby/2.3.0/rubygems/core_ext/kernel_require.rb:55:in `require'
	from /var/lib/gems/2.3.0/gems/backports-3.6.8/lib/backports/std_lib.rb:9:in `require_with_backports'
	from /home/gsteinbeis/security_monitor/3rd_party/ssh_scan/lib/ssh_scan/database.rb:1:in `<top (required)>'
	from /usr/lib/ruby/2.3.0/rubygems/core_ext/kernel_require.rb:55:in `require'
	from /usr/lib/ruby/2.3.0/rubygems/core_ext/kernel_require.rb:55:in `require'
	from /var/lib/gems/2.3.0/gems/backports-3.6.8/lib/backports/std_lib.rb:9:in `require_with_backports'
	from /home/gsteinbeis/security_monitor/3rd_party/ssh_scan/lib/ssh_scan/api.rb:6:in `<top (required)>'
	from /usr/lib/ruby/2.3.0/rubygems/core_ext/kernel_require.rb:55:in `require'
	from /usr/lib/ruby/2.3.0/rubygems/core_ext/kernel_require.rb:55:in `require'
	from /home/gsteinbeis/security_monitor/3rd_party/ssh_scan/lib/ssh_scan.rb:17:in `<top (required)>'
	from /usr/lib/ruby/2.3.0/rubygems/core_ext/kernel_require.rb:55:in `require'
	from /usr/lib/ruby/2.3.0/rubygems/core_ext/kernel_require.rb:55:in `require'
	from ./bin/ssh_scan:9:in `<main>'
</pre>

I am not familiar enough with ruby to debug this further. Switching to tag 0.0.16 resolved the issue.

**Comment by royalharsh** [comment]
Hi @gerhard-tinned ,

Try this: 
```bash
source ~/.rvm/scripts/rvm
rvm use 2.3.1
gem install bundler
bundle install
```

**Comment by gerhard-tinned** [comment]
The issue i reported started with commit 4c70e248b4148d7aaf4c7a4e5c6cae668f9dbc4e (Add mongodb as default database) but i will try your suggestion...

**Comment by gerhard-tinned** [comment]
> source ~/.rvm/scripts/rvm

... No such file or directory

> rvm use 2.3.1

-bash: rvm: command not found

> **sudo** gem install bundler

Fetching: bundler-1.14.5.gem (100%)
Successfully installed bundler-1.14.5
Parsing documentation for bundler-1.14.5
Installing ri documentation for bundler-1.14.5
Done installing documentation for bundler after 15 seconds
1 gem installed


> bundle install

...
Bundle complete! 8 Gemfile dependencies, 49 gems now installed.
Bundled gems are installed into ./vendor/bundle.

---

Issue is still the same.

**Comment by gerhard-tinned** [comment]
It seems as well that the following two files get modified as soon as the ssh_scan is executed the first time. So they show up on a `git status`.

<pre>
M	bin/ssh_scan_api_example_config.yml
M	bin/ssh_scan_worker_example_config.yml
</pre>

**Comment by gerhard-tinned** [comment]
It seems as if whatever causes the ssh_scan to fail was to do with the changes in 4c70e248b4148d7aaf4c7a4e5c6cae668f9dbc4e "Add mongodb as default database". As the commit before works, but this and later show the error. 

Any help or suggestions?

**Comment by claudijd** [comment]
@gerhard-tinned sorry about this, this is an artifact of us not really have a good handle on dependency management at the moment.  It's something we need to figure out before we release what's in master.  The latest release'd gem should be functional in this way.

**Comment by claudijd** [comment]
The issue should be resolved in master when #308 is fixed in master.

**Comment by gerhard-tinned** [comment]
The latest tag is 0.0.16 and this has the issue with the false positive in the compliant result as reported in #281 . That means the latest tag version (0.0.16) is also not an option as the issue #281 was (from what I understand and saw) only fixed in master branch.

Can I help in any way?

**Comment by claudijd** [comment]
@gerhard-tinned I spent some time today to break the API dependancies into their own repo.  Would you mind trying out the prerelease gem for 0.0.17 and letting me know if that resolves your issue?

https://rubygems.org/gems/ssh_scan/versions/0.0.17.pre

**Comment by gerhard-tinned** [comment]
I will checkout the version 0.0.17.pre as soon as I can and I will let you know the result.

**Comment by gerhard-tinned** [comment]
I tested the 0.0.17.pre tagged version. As soon as I switched to the 0.017.pre tag, the error message reported above was gone. 

I ran it against a few of my hosts with different policies. This version does not throw the error as listed below as well as it does show the expected result. The issue with the wrong compliant result as reported in a different ticket is gone. 

From my point of few, ready to deploy! 

Great work, thanks a lot!
